op-challenger/game/keccak/fetcher/fetcher_test.go

```diff
@@ -143,158 +278,238 @@ func TestFetchLeaves_SkipTxWithReceiptStatusFail(t *testing.T) {
-	if !ok {
+	proposalId := data[0]
+	proposal, ok := o.proposals[proposalId]
+	if !ok || !proposal.valid {
 		return nil, keccakTypes.InputData{}, contracts.ErrInvalidAddLeavesCall
 	}
-	uuid := ident.UUID
-	// WithUUID appends custom UUIDs to the tx data
-	if len(data) > 1 {
-		uuid = new(big.Int).SetBytes(data[1:])
-	}
-	return uuid, input, nil
+
+	return proposal.uuid, proposal.inputData, nil
 }
 
-type TxModifier func(tx *types.DynamicFeeTx) *ecdsa.PrivateKey
+type TxModifier func(tx *types.DynamicFeeTx)
 
-var ValidTx TxModifier = func(_ *types.DynamicFeeTx) *ecdsa.PrivateKey {
-	return privKey
+var ValidTx TxModifier = func(_ *types.DynamicFeeTx) {
+	// no-op
 }
 
 func WithToAddr(addr common.Address) TxModifier {
-	return func(tx *types.DynamicFeeTx) *ecdsa.PrivateKey {
+	return func(tx *types.DynamicFeeTx) {
 		tx.To = &addr
-		return privKey
 	}
 }
 
 func WithoutToAddr() TxModifier {
-	return func(tx *types.DynamicFeeTx) *ecdsa.PrivateKey {
+	return func(tx *types.DynamicFeeTx) {
 		tx.To = nil
-		return privKey
 	}
 }
 
-func WithUUID(uuid *big.Int) TxModifier {
-	return func(tx *types.DynamicFeeTx) *ecdsa.PrivateKey {
-		tx.Data = append(tx.Data, uuid.Bytes()...)
-		return privKey
+func (o *stubOracle) createValidProposal(input keccakTypes.InputData) *proposalConfig {
+	id := o.nextProposalId
+	o.nextProposalId++
+
+	proposal := &proposalConfig{
+		id:           id,
+		claimantAddr: ident.Claimant,
+		inputData:    input,
+		uuid:         ident.UUID,
+		valid:        true,
 	}
+	o.proposals[id] = proposal
+
+	return proposal
 }
 
-func WithInvalidData() TxModifier {
-	return func(tx *types.DynamicFeeTx) *ecdsa.PrivateKey {
-		tx.Data = []byte{}
-		return privKey
-	}
+type stubL1Source struct {
+	nextTxId uint64
+	// Map block number to tx
+	txs map[uint64]types.Transactions
+	// Map txHash to receipt
+	rcptStatus map[common.Hash]uint64
+	// Map txHash to logs
+	logs map[common.Hash][]*types.Log
 }
 
-func WithChainID(id *big.Int) TxModifier {
-	return func(tx *types.DynamicFeeTx) *ecdsa.PrivateKey {
-		tx.ChainID = id
-		return privKey
+func (s *stubL1Source) ChainID(_ context.Context) (*big.Int, error) {
+	return chainID, nil
+}
+
+func (s *stubL1Source) BlockByNumber(_ context.Context, number *big.Int) (*types.Block, error) {
+	txs, ok := s.txs[number.Uint64()]
+	if !ok {
+		return nil, errors.New("not found")
 	}
+	return (&types.Block{}).WithBody(txs, nil), nil
 }
 
-func WithPrivKey(key *ecdsa.PrivateKey) TxModifier {
-	return func(tx *types.DynamicFeeTx) *ecdsa.PrivateKey {
-		return key
+func (s *stubL1Source) TransactionReceipt(_ context.Context, txHash common.Hash) (*types.Receipt, error) {
+	rcptStatus, ok := s.rcptStatus[txHash]
+	if !ok {
+		rcptStatus = types.ReceiptStatusSuccessful
+	} else if rcptStatus == MissingReceiptStatus {
+		return nil, errors.New("not found")
 	}
+
+	logs, ok := s.logs[txHash]
+	if !ok {
+		logs = make([]*types.Log, 0)
+	}
+	return &types.Receipt{Status: rcptStatus, Logs: logs}, nil
```

Comment:
We can just use nil as an empty slice here.
```suggestion
	logs := s.logs[txHash]
	return &types.Receipt{Status: rcptStatus, Logs: logs}, nil
```